werkzeug/routing.py

```diff
@@ -1498,7 +1498,7 @@ def match(self, path_info=None, method=None, return_rule=False,
         """
         self.map.update()
         if path_info is None:
-            path_info = self.path_info
+            path_info = self.path_info or u'/'
```

Comment:
Perhaps this logic would be better suited for `bind`, such that `self.path_info` is not None?

Something like:

```diff
diff --git a/werkzeug/routing.py b/werkzeug/routing.py
index 82cf793..888ac9b 100644
--- a/werkzeug/routing.py
+++ b/werkzeug/routing.py
@@ -1232,6 +1232,8 @@ class Map(object):
             subdomain = self.default_subdomain
         if script_name is None:
             script_name = '/'
+        if path_info is None:
+            path_info = '/'
         server_name = _encode_idna(server_name)
         return MapAdapter(self, server_name, script_name, subdomain,
                           url_scheme, path_info, default_method, query_args)
@@ -1498,7 +1500,7 @@ class MapAdapter(object):
         """
         self.map.update()
         if path_info is None:
-            path_info = self.path_info or u'/'
+            path_info = self.path_info
         else:
             path_info = to_unicode(path_info, self.map.charset)
         if query_args is None:
```